master/pkg/model/task.go

```diff
@@ -260,52 +261,54 @@ type TaskLog struct {
 	Timestamp   *time.Time `db:"timestamp" json:"timestamp"`
 	Level       *string    `db:"level" json:"level"`
 	Log         string     `db:"log" json:"log"`
-	FlatLog     string     `db:"message" json:"message,omitempty"`
 	Source      *string    `db:"source" json:"source,omitempty"`
 	StdType     *string    `db:"stdtype" json:"stdtype,omitempty"`
 }
 
-// RFC3339MicroTrailingZeroes unlike time.RFC3339Nano is a time format specifier that preserves
-// trailing zeroes.
-const RFC3339MicroTrailingZeroes = "2006-01-02T15:04:05.000000Z07:00"
+const (
+	// RFC3339MicroTrailingZeroes unlike time.RFC3339Nano is a time format specifier that preserves
+	// trailing zeroes.
+	RFC3339MicroTrailingZeroes = "2006-01-02T15:04:05.000000Z07:00"
+	// containerIDMaxLength is the max display length for a container ID in logs.
+	containerIDMaxLength = 8
+)
 
 // Message resolves the flat version of the log that UIs have shown historically.
 // TODO(task-unif): Should we just.. stop doing this? And send the log as is and let the
 // UIs handle display (yes, IMO).
 func (t *TaskLog) Message() string {
-	if t.FlatLog != "" {
-		return t.FlatLog
-	}
+	var builder strings.Builder
 
-	var timestamp string
+	// e.g., "[2022-03-02T02:15:18.299569Z]"
 	if t.Timestamp != nil {
-		timestamp = t.Timestamp.Format(RFC3339MicroTrailingZeroes)
+		builder.WriteString(fmt.Sprintf("[%s]", t.Timestamp.Format(RFC3339MicroTrailingZeroes)))
 	} else {
-		timestamp = defaultTaskLogTime
+		builder.WriteString(fmt.Sprintf("[%s]", defaultTaskLogTime))
 	}
 
-	// This is just to match postgres.
-	const containerIDMaxLength = 8
-	var containerID string
+	// e.g., " f6114bb3"
 	if t.ContainerID != nil && *t.ContainerID != "" {
-		containerID = *t.ContainerID
+		containerID := *t.ContainerID
 		if len(containerID) > containerIDMaxLength {
 			containerID = containerID[:containerIDMaxLength]
 		}
-		containerID = fmt.Sprintf("%s ", containerID)
+		builder.WriteString(fmt.Sprintf(" %s", containerID))
 	}
 
-	var rankID string
+	// e.g., " [rank=1]"
 	if t.RankID != nil {
-		rankID = fmt.Sprintf("[rank=%d] ", *t.RankID)
+		builder.WriteString(fmt.Sprintf(" [rank=%d] ", *t.RankID))
 	}
 
-	var level string
+	builder.WriteString(" ||")
+
+	// e.g., " INFO"
 	if t.Level != nil {
-		level = fmt.Sprintf("%s: ", *t.Level)
+		builder.WriteString(fmt.Sprintf(" %s", *t.Level))
 	}
 
-	return fmt.Sprintf("[%s] %s%s|| %s %s", timestamp, containerID, rankID, level, t.Log)
```

Comment:
i prefer the builder because i was trying to understand where an extra space was coming from. when I wrote it i _didnt_ prefer the builder. i'm fine either way.

Reply:
Third alternative: Build up `pieces []string` and end with `strings.Join(pieces, " ")`. I think that makes all the space insertion handling the clearest (i.e., you don't have to consider it at all).

Reply:
i prefer this, maybe?

Reply:
this doesn't matter to much because i want to deprecate v1 trial logs and just make a v2 that returns the structured form and let the webui do this